game/player.go

```diff
@@ -153,6 +154,43 @@ func (p *Player) RemoveFromBoard(perm *Permanent) {
 	}
 }
 
+// Returns possible actions when we can activate cards on he board.
+func (p *Player) ActivatedAbilityActions(allowSorcerySpeed bool, forHuman bool) []*Action {
+	permNames := make(map[CardName]bool)
+	answer := []*Action{}
+
+	for _, perm := range p.Board { // TODO could be opponent's board for some actions, e.g. Warmonger
+		// Don't re-check playing duplicate actions
+		if permNames[perm.Name] {
+			continue
+		}
+		if perm.DidActivate {
+			continue
+		}
+		permNames[perm.Name] = true
+		if perm.ActivatedAbility != nil {
+			effect := perm.ActivatedAbility
+			if effect.TargetType.Type == Creature { // TODO lands etc
```

Comment:
there must be some way to make this less of a pyramid-looking function. maybe a helper function named "LegalTargets" of a TargetType (or Selector)